### ISSUE TITLE
Have tun support multiple address

### ISSUE THREAD

**Issue by ibigbug** [Watfaq/clash-rs#579]
The rust tun only take one address now. We'll need both v4 and v6 address to add static route on windows

**Comment by VendettaReborn** [comment]
rust-tun and wintun don't support settings of ipv6 addr yet, and since hard to push this feature to upstreams(at least might take plenty of time), we may extend the `Device` trait now, maybe in our repo, maybe in a separate crate?
here is the work-around on each platform

- windows: https://github.com/firezone/firezone/blob/3d8ed7f10ed596bb7fb3ca376888eb10a267bc50/rust/connlib/tunnel/src/device_channel/tun_windows.rs#L90
- macos: https://github.com/meh/rust-tun/pull/91/commits/13ae6cec497d2cf63a9d7883969109cba624f57c
- linux: https://github.com/rust-netlink/rtnetlink/blob/5fca904b11ba2535fdfac30bf729aa8c10c34c0d/examples/add_address.rs#L32

**Comment by ibigbug** [comment]
yeah wintun is a bit weird it does take v6 address but doesn't expose it https://github.com/nulldotblack/wintun/blob/main/src/adapter.rs#L306

i think we can fork the rust-tun(or tun2?) and add v6 support in a separate repo and try to add it back to the upstream after we validated it's working

**Comment by VendettaReborn** [comment]
oh, i forgot that we are currently using the forked rust-tun. yeah, that's workable

**Comment by xmh0511** [comment]
Welcome to use [tun-rs](https://crates.io/crates/tun-rs), which is actively maintained.
